service/gcs/bridge/bridge.go

```diff
@@ -566,26 +566,44 @@ func (b *Bridge) execProcess(w ResponseWriter, r *Request) {
 	}
 
 	var pid int
+	// If this is the exec of the init process for V1 we need to return the
+	// error of the exec previous to any ContainerExited notification. So we
+	// signal when we are done writing to the bridge.
+	var execInitErrorDone chan<- struct{}
+
+	defer func() {
+		if execInitErrorDone != nil {
+			execInitErrorDone <- struct{}{}
+		}
+	}()
+
 	var err error
 	if params.IsExternal {
+		logrus.Debugf("bridge:execProcess: IsExternal Calling b.coreint.RunExternalProcess")
 		pid, err = b.coreint.RunExternalProcess(params, conSettings)
 	} else if params.SchemaVersion.Cmp(prot.SchemaVersion{Major: 2, Minor: 0}) >= 0 {
 		var c *gcspkg.Container
 		c, err = b.hostState.GetContainer(request.ContainerID)
 		if err == nil {
 			if params.OCIProcess == nil {
+				logrus.Debugf("bridge::execProcess: v2 No OCI Process - calling start")
 				pid, err = c.Start(conSettings)
 			} else {
+				logrus.Debug("bridge::execProcess: v2 Have OCI Process - calling ExecProcess")
 				pid, err = c.ExecProcess(params.OCIProcess, conSettings)
 			}
 		}
 	} else {
-		pid, err = b.coreint.ExecProcess(request.ContainerID, params, conSettings)
+		logrus.Debug("bridge::execProcess: Calling b.coreint.ExecProcess")
+		pid, execInitErrorDone, err = b.coreint.ExecProcess(request.ContainerID, params, conSettings)
 	}
 
 	if err != nil {
+		logrus.Errorf("bridge::execProcess Error from ExecProcess: %s", err)
 		w.Error(request.ActivityID, err)
 		return
+	} else {
```

Comment:
There is a return on error so dont need else.

Reply:
Fixed